src/app/view/components/dot-device-selector/dot-device-selector.component.ts

```diff
@@ -21,14 +21,19 @@ export class DotDeviceSelectorComponent implements OnInit {
     ngOnInit() {
         this.options = this.dotMessageService
             .getMessages(['editpage.viewas.default.device'])
-            .mergeMap(() =>
-                this.dotDevicesService
-                    .get()
-                    .pipe(
-                        map((devices: DotDevice[]) =>
-                            this.setOptions(this.dotMessageService.get('editpage.viewas.default.device'), devices)
+            .pipe(
+                mergeMap(() =>
+                    this.dotDevicesService
+                        .get()
+                        .pipe(
+                            flatMap((devices: DotDevice[]) => devices),
```

Comment:
Are you sure that we need a flatMap here?

Reply:
We need to convert the data so the filter operator can work with the data correctly